install.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+# This is an install script for notes
+
+function assertInstalled() {
+    for var in "$@"; do
+        if ! which $var &> /dev/null; then
+            echo "$var is required but not installed, exiting."
+            exit 1
+        fi
+    done
+}
+
+# Yay, Echo self documents! :D
+echo "Checking for root..."
+    [ "$(whoami)" != "root" ] && exec sudo -- "$0" "$@"
+
+# Make sure we have everything actually installed that we need to install this.
+echo "Checking for Dependencies..."
+    assertInstalled bash curl tar mktemp install make
```

Comment:
I think this (and the root check too probably) should live in the makefile instead.

The `install` assert definitely, because that's only used in the Makefile, and what we need installed in there might change. It'd be good to have things specified in the same file they're used, to ensure they're consistent and make it easy to remember to keep them up to date. The root check is less important, but this script doesn't *require* root afaict, and if we can move as much as possible into the Makefile to keep things simpler rather than spread over two files that'd be great.

That would make this script responsible for fetching and starting the Makefile, and getting the minimum required for that to happen, and the Makefile responsible for all the rest.